### PR TITLE
Update dig to allow dropping an extra point before placing second point.

### PR DESCRIPTION
**Pull request by johncosker**
Demo3

A reddit comment got me thinking and I realized a nice UX feature is being able to use the `v` shortcut to drop an extra point for a curve whil placing the second point of a line. 

It actually works really well and I think is really intuitive.

### CODE REVIEW

File: gui/dig.lua [hunk 1]
```diff
@@ -247,8 +247,13 @@ function GenericOptionsPanel:init()
             end,
             show_tooltip = true,
             on_activate = function()
-                self.dig_panel.placing_extra.active = true
-                self.dig_panel.placing_extra.index = #self.dig_panel.extra_points + 1
+                if self.dig_panel.mark1 and self.dig_panel.mark2 then
+                    self.dig_panel.placing_extra.active = true
+                    self.dig_panel.placing_extra.index = #self.dig_panel.extra_points + 1
+                elseif self.dig_panel.mark1 and not self.dig_panel.mark2 then
+                    local mouse_pos = dfhack.gui.getMousePos()
```
Comment (myk002) [inline gui/dig.lua:254]: be sure to check mouse_pos for nil if the player hits `v` while the mouse is out of bounds

File: gui/dig.lua [hunk 2]
```diff
@@ -247,8 +247,13 @@ function GenericOptionsPanel:init()
...
+                    self.dig_panel.extra_points[#self.dig_panel.extra_points + 1] = { x = mouse_pos.x, y = mouse_pos.y }
```
Comment (myk002) [inline gui/dig.lua:255]:
```suggestion
                    table.insert(self.dig_panel.extra_points, { x = mouse_pos.x, y = mouse_pos.y })
```

Comment (myk002) [reply gui/dig.lua:255]: not a required change, but slightly more concise

**Review by myk002 — COMMENTED** [review]
changelog entry plz

**Comment by johncosker** [comment]
@myk002 would this be a dev changelog or normal?

**Comment by myk002** [comment]
Dev changelog, since it's changing something that hasn't seen a "full" release yet

**Review by myk002 — APPROVED** [review]

**Pull request merged**